### PR TITLE
Make application search more user friendly

### PR DESCRIPTION
**Pull request by Lonami**
The old algorithm was really confusing and had zero comments. This new algorithms is easy to follow and yields better results that are more friendly to the user, for instance, looking for "ffox" will match "firefox" while the old algorithm didn't.

### CODE REVIEW

**Comment by divadsn** [comment]
Fix Codacy please :)

**Comment by Lonami** [comment]
There we go.

File: app/src/main/java/ch/deletescape/lawnchair/allapps/UnicodeStrippedAppSearchAlgorithm.java [hunk 1]
```diff
@@ -22,11 +22,6 @@ protected boolean matches(AppInfo info, String query) {
 
         String title = UnicodeFilter.filter(info.title.toString().toLowerCase());
         String strippedQuery = UnicodeFilter.filter(query.trim());
-        int queryLength = strippedQuery.length();
-
-        if (title.length() < queryLength || queryLength <= 0) {
-            return false;
-        }
```
Comment (Lonami) [inline app/src/main/java/ch/deletescape/lawnchair/allapps/UnicodeStrippedAppSearchAlgorithm.java]: All this is done inside `super.matches` so it's a bit redundant.

**Review by divadsn — APPROVED** [review]
Look good, gonna merge after testing.

**Comment by Lonami** [comment]
Nice, thanks!

**Comment by AxiK101** [comment]
👍

**Pull request merged**